x-pack/plugin/data-streams/src/main/java/org/elasticsearch/xpack/datastreams/DataStreamIndexSettingsProvider.java

```diff
@@ -30,41 +30,49 @@
     public Settings getAdditionalIndexSettings(
         String indexName,
         String dataStreamName,
+        ComposableIndexTemplate matchingTemplate,
         Metadata metadata,
         long resolvedAt,
         Settings allSettings
     ) {
         if (dataStreamName != null) {
-            IndexMode indexMode = Optional.ofNullable(allSettings.get(IndexSettings.MODE.getKey()))
-                .map(value -> IndexMode.valueOf(value.toUpperCase(Locale.ROOT)))
-                .orElse(IndexMode.STANDARD);
-            if (indexMode == IndexMode.TIME_SERIES) {
-                TimeValue lookAheadTime = allSettings.getAsTime(
-                    IndexSettings.LOOK_AHEAD_TIME.getKey(),
-                    IndexSettings.LOOK_AHEAD_TIME.getDefault(allSettings)
-                );
+            DataStream dataStream = metadata.dataStreams().get(dataStreamName);
+            IndexMode indexMode;
+            if (dataStream != null) {
+                indexMode = dataStream.getIndexMode();
+            } else {
+                indexMode = matchingTemplate.getDataStreamTemplate().getIndexMode();
+            }
+            if (indexMode != null) {
                 Settings.Builder builder = Settings.builder();
-                DataStream dataStream = metadata.dataStreams().get(dataStreamName);
-                Instant start;
-                if (dataStream == null) {
-                    start = Instant.ofEpochMilli(resolvedAt).minusMillis(lookAheadTime.getMillis());
-                } else {
-                    IndexMetadata currentLatestBackingIndex = metadata.index(dataStream.getWriteIndex());
-                    if (currentLatestBackingIndex.getSettings().hasValue(IndexSettings.TIME_SERIES_END_TIME.getKey()) == false) {
-                        throw new IllegalStateException(
-                            String.format(
-                                Locale.ROOT,
-                                "backing index [%s] in tsdb mode doesn't have the [%s] index setting",
-                                currentLatestBackingIndex.getIndex().getName(),
-                                IndexSettings.TIME_SERIES_START_TIME.getKey()
-                            )
-                        );
+                builder.put(IndexSettings.MODE.getKey(), indexMode.name().toLowerCase(Locale.ROOT));
+
+                if (indexMode == IndexMode.TIME_SERIES) {
+                    TimeValue lookAheadTime = allSettings.getAsTime(
+                        IndexSettings.LOOK_AHEAD_TIME.getKey(),
+                        IndexSettings.LOOK_AHEAD_TIME.getDefault(allSettings)
+                    );
```

Comment:
Can we use `IndexSettings.LOOK_AHEAD_TIME.get(allSettings)` which should resolve the default value appropriately here instead?